### PR TITLE
Store size limit to 1GiB

### PR DESCRIPTION
**Pull request by sduchesneau**
* store total size is now limited to 1GiB.
* TotalSizeBytes is calculated on load (during the unmarshal) and modified on every kv operation to avoid recalculation

### CODE REVIEW

**Comment by maoueh** [comment]
@sduchesneau Please add changelog entry for this change

File: storage/store/merge.go [hunk 1]
```diff
@@ -33,28 +48,28 @@ func (b *baseStore) Merge(kvPartialStore *PartialKV) error {
 	switch b.updatePolicy {
 	case pbsubstreams.Module_KindStore_UPDATE_POLICY_SET:
 		for k, v := range kvPartialStore.kv {
-			b.kv[k] = v
+			b.setKV(k, v)
```
Comment (abourget) [inline storage/store/merge.go:51]: `updateKV` to clarify intent, different from `setNewKV` ? Someone could easily mix `setKV` and `setNewKV` not knowing the difference.

File: storage/store/merge.go [hunk 2]
```diff
@@ -12,6 +12,21 @@ import (
 	pbsubstreams "github.com/streamingfast/substreams/pb/sf/substreams/v1"
 )
 
+func (b *baseStore) setKV(k string, v []byte) {
+	if prev, ok := b.kv[k]; ok {
```
Comment (abourget) [inline storage/store/merge.go:16]: This does add a _second_ lookup though; 
The caller of `setKV` often did it itself.

We can live with it for now I think.

**Comment by abourget** [comment]
Looked good in general otherwise ;)

**Pull request merged**